src/openvic-simulation/modifier/ModifierManager.cpp

```diff
@@ -686,181 +659,140 @@ bool ModifierManager::parse_scripts(DefinitionManager const& definition_manager)
+			)(value);
 
-node_callback_t ModifierManager::expect_whitelisted_modifier_value(
-	callback_t<ModifierValue&&> modifier_callback, Modifier::modifier_type_t type, string_set_t const& whitelist
-) const {
-	return expect_whitelisted_modifier_value_and_default(modifier_callback, type, whitelist, key_value_invalid_callback);
-}
+			ret &= _add_flattened_modifier_cb(modifier_value, key, faction_identifier, value_node);
 
-node_callback_t ModifierManager::expect_modifier_value_and_key_map_and_default(
-	callback_t<ModifierValue&&> modifier_callback, Modifier::modifier_type_t type, key_value_callback_t default_callback,
-	key_map_t&& key_map
-) const {
-	return [this, modifier_callback, type, default_callback, key_map = std::move(key_map)](
-		ast::NodeCPtr node
-	) mutable -> bool {
-		bool ret = expect_modifier_value_and_default(
-			modifier_callback, type, dictionary_keys_callback(key_map, default_callback)
-		)(node);
+			return ret;
+		}
 
-		ret &= check_key_map_counts(key_map);
+		if (strings_equal_case_insensitive(key, "farm_rgo_eff")) {
+			return _add_modifier_cb(modifier_value, modifier_effect_cache.farm_rgo_throughput_global, value)
+				&& _add_modifier_cb(modifier_value, modifier_effect_cache.farm_rgo_output_global, value);
+		}
+		
+		if (strings_equal_case_insensitive(key, "mine_rgo_eff")) {
+			return _add_modifier_cb(modifier_value, modifier_effect_cache.mine_rgo_throughput_global, value)
+				&& _add_modifier_cb(modifier_value, modifier_effect_cache.mine_rgo_output_global, value);
+		}
 
-		return ret;
+		return _expect_modifier_effect_with_fallback(
+			technology_modifier_effects,
+			modifier_value,
+		_expect_shared_tech_country_modifier_effect(modifier_value)
+		)(key, value);
 	};
 }
-
-node_callback_t ModifierManager::expect_modifier_value_and_key_map(
-	callback_t<ModifierValue&&> modifier_callback, Modifier::modifier_type_t type, key_map_t&& key_map
+NodeTools::key_value_callback_t ModifierManager::expect_unit_terrain_modifier(
+	ModifierValue& modifier_value,
+	const std::string_view terrain_type_identifier
 ) const {
-	return expect_modifier_value_and_key_map_and_default(
-		modifier_callback, type, key_value_invalid_callback, std::move(key_map)
+	return [this, &modifier_value, &terrain_type_identifier](const std::string_view key, const ast::NodeCPtr value) -> bool {
+		const std::string flat_identifier = get_flat_identifier(key, terrain_type_identifier);
+		ModifierEffect const* effect = unit_terrain_modifier_effects.get_item_by_identifier(flat_identifier);
+		if (effect == nullptr) {
+			return key_value_invalid_callback(flat_identifier, value);
+		}
+		return _add_modifier_cb(modifier_value, effect, value);
+	};
+}
+NodeTools::key_value_callback_t ModifierManager::expect_base_country_modifier(ModifierValue& modifier_value) const {
+	return _expect_modifier_effect_with_fallback(
+		base_country_modifier_effects,
+		modifier_value,
+	_expect_shared_tech_country_modifier_effect(modifier_value)
```

Comment:
Formatting: this should be indented inline with the other arguments above it, same for all the other functions below